src/sharechain/in_memory.rs

```
@@ -115,29 +115,30 @@ impl InMemoryShareChain {
         last_level
             .blocks
             .iter()
-            .max_by(|block1, block2| block1.height().cmp(&block2.height()))
+            .max_by(|block1, block2| block1.height.cmp(&block2.height))
             .cloned()
     }
 
     /// Calculates block difficulty based on it's pow algo.
     fn block_difficulty(&self, block: &Block) -> Result<u64, Error> {
-        match block.original_block_header().pow.pow_algo {
+        match block.original_block_header.pow.pow_algo {
             PowAlgorithm::RandomX => {
                 if let Some(params) = &self.block_validation_params {
                     let difficulty = randomx_difficulty(
-                        block.original_block_header(),
+                        &block.original_block_header,
                         params.random_x_factory(),
                         params.genesis_block_hash(),
                         params.consensus_manager(),
                     )
                     .map_err(Error::RandomXDifficulty)?;
                     Ok(difficulty.as_u64())
                 } else {
-                    Ok(0)
+                    panic!("No params provided for RandomX difficulty calculation!");
```

Why not return an error here?